util/ranger/detacher.go

```diff
@@ -706,9 +709,10 @@ func removeAccessConditions(conditions, accessConds []expression.Expression) []e
 
 // ExtractAccessConditionsForColumn extracts the access conditions used for range calculation. Since
 // we don't need to return the remained filter conditions, it is much simpler than DetachCondsForColumn.
-func ExtractAccessConditionsForColumn(conds []expression.Expression, uniqueID int64) []expression.Expression {
+func ExtractAccessConditionsForColumn(conds []expression.Expression, col *expression.Column) []expression.Expression {
 	checker := conditionChecker{
-		colUniqueID: uniqueID,
+		colUniqueID: col.UniqueID,
+		checkerCol:  col,
```

Comment:
Can we remove `colUniqueID`?

Reply:
IMO, this can keep compatibility. If we don't want to compare virtual expressions in some scenarios, we can just use colUniqueID.

Reply:
Oh, I take it wrong. Done in the new commit